tests/unit/cli_test.py

```diff
@@ -127,7 +127,10 @@ def test_run_interactive_passes_logs_false(self, mock_pseudo_terminal, mock_run_
         _, _, call_kwargs = mock_run_operation.mock_calls[0]
         assert call_kwargs['logs'] is False
 
-    def test_run_service_with_restart_always(self):
+    @pytest.mark.xfail(IS_WINDOWS_PLATFORM, reason="requires dockerpty")
+    @mock.patch('compose.cli.main.RunOperation', autospec=True)
+    @mock.patch('compose.cli.main.PseudoTerminal', autospec=True)
+    def test_run_service_with_restart_always(self, mock_pseudo_terminal, mock_run_operation):
```

Comment:
That existing test was calling with the now-invalid combo of -d and --rm. This path took advantage of the early exit from `run_one_off_container` and so did not need to worry about mocking the TTY behaviors.

This changes the test to :
* use -d: false (since the --rm is key to the behaviors under test)
* uses mocked versions of the Run/Ptty (as that path is now exercised)